### PR TITLE
Telemetry

### PR DESCRIPTION
**Pull request by eu9ene**
Basic telemetry integration:
- v1 schema (everything is in one ping which is not too bad because it reduces the number of pings, simplifies sending and doesn't require expensive joins on analysis)
- simple and lightweight glean client, that imitates behavior of glean.js 0.15.0 ( the one that was integrated in the legacy extension)
- the most important in-page translation metrics are collected

Out of scope for this PR:
- #42 
- #43 
- #37 
- #45 
- edge cases when ping is not sent (browser event wasn't triggered) - to find

### CODE REVIEW

File: extension/mediator.js [hunk 1]
```diff
@@ -27,7 +32,16 @@ class Mediator {
     }
 
     init() {
-        browser.runtime.sendMessage({ command: "monitorTabLoad" });
+        this.telemetry.loadSchema()
+            .then(() => {
+                browser.runtime.sendMessage({ command: "monitorTabLoad" })
```
Comment (eu9ene) [inline extension/mediator.js:37]: sometimes mediator.start is called before telemetry schema is loaded on page refresh. I don't understand why this happens, probably old subscription to "responseMonitorTabLoad" in background script survives the refresh.

File: extension/controller/backgroundScript.js [hunk 2]
```diff
@@ -7,6 +7,10 @@
  * this script does not have access to the page content, since it runs in the
  * extension's background process.
  */
+
+let cachedEnvInfo = null;
+console.debug("bg script loaded");
```
Comment (andrenatal) [inline extension/controller/backgroundScript.js:12]: can we remove this?

File: extension/mediator.js [hunk 3]
```diff
@@ -28,12 +33,19 @@ class Mediator {
...
+
+    // the page is closed or infobar is closed manually
+    closeSession() {
+        this.telemetry.submit("custom");
     }
 
     // main entrypoint to handle the extension's load
     start(tabId) {
-
+        console.debug("starting mediator");
```
Comment (andrenatal) [inline extension/mediator.js:46]: remove this

File: extension/mediator.js [hunk 4]
```diff
@@ -152,19 +199,24 @@ class Mediator {
...
+    // eslint-disable-next-line max-lines-per-function
     bgScriptsMessageListener(message) {
         switch (message.command) {
             case "responseMonitorTabLoad":
                 this.start(message.tabId);
                 break;
+            case "telemetryInfoLoaded":
+                this.telemetry.setBrowserEnv(message.env);
+                this.translationTelemetry.recordEnvironment(message.env);
+                this.translationTelemetry.recordVersions("0.5.0", "?", "?");
```
Comment (andrenatal) [inline extension/mediator.js:211]: is this the addon version? maybe is better to read it from the manifest.

**Review by andrenatal — APPROVED** [review]
I didn't test it but it's looking good. For the next patches is better to have smaller PRs

**Comment by andrenatal** [comment]
Just make sure to read the addon version from the manifest to automate that.

File: extension/mediator.js (hunk 4 above)
Comment (eu9ene) [reply extension/mediator.js:210]: I wanted to do this in #42

File: extension/mediator.js (hunk 3 above)
Comment (eu9ene) [reply extension/mediator.js]: removed

File: extension/controller/backgroundScript.js (hunk 2 above)
Comment (eu9ene) [reply extension/controller/backgroundScript.js]: removed

File: extension/mediator.js (hunk 4 above)
Comment (eu9ene) [reply extension/mediator.js]: done

**Review by andrenatal — APPROVED** [review]

**Pull request merged**